test/clustertest/tests/MultipleLeaderSyncTest.cpp

```diff
@@ -9,19 +9,46 @@ struct MultipleLeaderSyncTest : tpunit::TestFixture {
     // Create a bunch of trivial write commands.
     void runTrivialWrites(int writeCount, BedrockTester& node) {
         int count = 0;
+
+        SData genericRequest("Query");
+        genericRequest["query"] = "UPDATE test SET value=value + 1 WHERE id=12345;";
+        genericRequest["connection"] = "forget";
+        genericRequest["writeConsistency"] = "ASYNC";
+        vector<SData> genericRequests;
+        for (int i = 0; i < 10; i++) {
+            genericRequests.push_back(genericRequest);
+        }
+
         while (count <= writeCount) {
             SData request;
             request.methodLine = "Query";
             if (count == 0) {
                 request["query"] = "INSERT OR REPLACE INTO test (id, value) VALUES(12345, 1 );";
                 node.executeWaitVerifyContent(request, "200");
+                count += 1;
             } else {
-                request["query"] = "UPDATE test SET value=value + 1 WHERE id=12345;";
-                request["connection"] = "forget";
-                node.executeWaitVerifyContent(request, "202");
+                node.executeWaitMultipleData(genericRequests);
+                count += 10;
+            }
+        }
+    }
+
+    bool waitForCommit(BedrockTester& node, uint64_t minCommitCount, uint64_t timeoutUS = 60'000'000) {
+        uint64_t start = STimeNow();
+        while (STimeNow() < start + timeoutUS) {
+            try {
+                string result = SParseJSONObject(node.executeWaitVerifyContent(SData("Status"), "200", true))["commitCount"];
+
+                // if the value matches, return, otherwise wait
+                if (SToUInt64(result) >= minCommitCount) {
+                    return true;
+                }
+            } catch (...) {
```

Comment:
It could probably just catch `SException`. If `executeWaitVerifyContent` fails it will throw, and it can do that if the node is offline. In that case we want to wait for it to come back up.

Can we wait for this and do it in a subsequent change? It should only affect Bedrock (nothing else uses `waitForCommit`) but I don't want to change the bedrock PR that three other PRs depend on and have already been tested against if we don't have to.

Reply:
Followup PR sounds fine